widgets/src/c3/Gauge.js

```diff
@@ -0,0 +1,43 @@
+(function (root, factory) {
+    if (typeof define === "function" && define.amd) {
+        define(["./Common"], factory);
+    } else {
+        root.Gauge = factory(root.Common);
+    }
+}(this, function (Common) {
+    function Gauge(target) {
+        Common.call(this);
+
+        this._class = "c3_gauge";
+        this._type = "gauge";
+        this._config.data = {
+            columns: []
+        };
+    };
+    Gauge.prototype = Object.create(Common.prototype);
+
+    Gauge.prototype.palette = function (_) {
+        if (!arguments.length) return this._palette;
+        this._config.color.pattern = _;
```

Comment:
I agree, while I do have some helper functions in https://github.com/hpcc-systems/Visualization/blob/master/widgets/src/common/Palette.js

I think they should probably be wrapped up into a single Palette Object and then that can be shared at the Dashboard level (maybe one of each type?), which would ensure that each individual widget is working from the same color pool.

Then at the widget level you could simply create a new instance of the Palette object specifically for it (if you wanted a different color pool)?